packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -798,165 +804,162 @@ export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
-                          onQuerySubmit
-                        );
-
-                        if (!isCodeEditorExpanded) {
-                          editor.onDidContentSizeChange((e) => {
-                            // @ts-expect-error the property _oldContentHeight exists on the event object received but
-                            // is not available on the type definition
-                            if (e.contentHeight !== e._oldContentHeight) {
-                              updateHeight(editor);
-                            }
-                          });
+                    editor.onDidChangeModelContent((e) => {
+                      if (updateLinesFromModel) {
+                        lines = model?.getLineCount() || 1;
+                      }
+                      const currentPosition = editor.getPosition();
+                      const content = editorModel.current?.getValueInRange({
+                        startLineNumber: 0,
+                        startColumn: 0,
+                        endLineNumber: currentPosition?.lineNumber ?? 1,
+                        endColumn: currentPosition?.column ?? 1,
+                      });
+                      if (content) {
+                        codeRef.current = content || editor.getValue();
+                      }
+                    });
+
+                    // this is fixing a bug between the EUIPopover and the monaco editor
+                    // when the user clicks the editor, we force it to focus and the onDidFocusEditorText
+                    // to fire, the timeout is needed because otherwise it refocuses on the popover icon
+                    // and the user needs to click again the editor.
+                    // IMPORTANT: The popover needs to be wrapped with the EuiOutsideClickDetector component.
+                    editor.onMouseDown(() => {
+                      setTimeout(() => {
+                        editor.focus();
+                      }, 100);
+                    });
+
+                    editor.onDidFocusEditorText(() => {
+                      onEditorFocus();
+                    });
+
+                    editor.onKeyDown(() => {
+                      onEditorFocus();
+                    });
+
+                    // on CMD/CTRL + Enter submit the query
+                    editor.addCommand(
+                      // eslint-disable-next-line no-bitwise
+                      monaco.KeyMod.CtrlCmd | monaco.KeyCode.Enter,
+                      onQuerySubmit
+                    );
+
+                    editor.onDidLayoutChange((layoutInfoEvent) => {
+                      if (layoutInfoEvent.contentWidth !== measuredEditorWidth) {
+                        setMeasuredEditorWidth(layoutInfoEvent.contentWidth);
+                      }
+                    });
+
+                    if (!isCodeEditorExpanded) {
+                      editor.onDidContentSizeChange((e) => {
+                        if (e.contentHeightChanged) {
```

Comment:
I'm assuming this is a safe change, but maybe there are cases where `e.contentHeightChanged` is false, and `e.contentHeight !== e._oldContentHeight`?